substrate/frame/nomination-pools/src/migration.rs

```diff
@@ -107,6 +107,98 @@ pub mod unversioned {
 			Ok(())
 		}
 	}
+
+	/// Migrate existing pools from [`adapter::StakeStrategyType::Transfer`] to
+	/// [`adapter::StakeStrategyType::Delegate`].
+	///
+	/// Note: This only migrates the pools, the members are not migrated. They can use the
+	/// permission-less [`Pallet::claim_delegation()`] to migrate their funds.
+	///
+	/// This migration does not break any existing pool storage item, does not need to happen in any
+	/// sequence and hence can be applied unversioned on a production runtime.
+	pub struct DelegationStakeMigration<T>(sp_std::marker::PhantomData<T>);
+
+	// FIXME(ank4n) convert to MBM.
```

Comment:
There are currently 224 pools in polkadot and 179 in Kusama so its probably totally fine to do this in single block.